### PR TITLE
Cycle model

### PR DESCRIPTION
**Pull request by ngarivictor**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/hospitalcore/InventoryCommonService.java [hunk 1]
```diff
@@ -59,7 +59,9 @@ public interface InventoryCommonService extends OpenmrsService{
 
     public void voidRegimen(Regimen regimen) throws APIException;
 
-	List<Regimen> getRegimens (boolean voided);
+    public void createCycle(Cycle cycle) throws  APIException;
```
Comment (stansL) [inline api/src/main/java/org/openmrs/module/hospitalcore/InventoryCommonService.java]: we still need the getRegimens method that returns a list of Regimen objects. Noticed this has been deleted

File: api/src/main/java/org/openmrs/module/hospitalcore/db/InventoryCommonDAO.java [hunk 2]
```diff
@@ -55,4 +55,7 @@ public interface InventoryCommonDAO {
     Regimen createRegimen(Regimen regimen);
     Regimen updateRegimen(Regimen regimen);
     void voidRegimen(Regimen regimen);
+
```
Comment (stansL) [inline api/src/main/java/org/openmrs/module/hospitalcore/db/InventoryCommonDAO.java:62]: We also need a voidCycle method
```suggestion
void voidCycle(Cycle cycle) throws APIException;
```

File: api/src/main/resources/liquibase.xml [hunk 3]
```diff
@@ -1710,6 +1710,23 @@
...
+            <dbms type="mysql"/>
+        </preConditions>
+        <comment>Add table cycle</comment>
+        <sql>
+            CREATE TABLE IF NOT EXISTS `cycle` (
+            `id` int(11) NOT NULL AUTO_INCREMENT,
+            `summary_notes` varchar(255) DEFAULT NULL,
+            `active` tinyint(1) NOT NULL DEFAULT '0',
+            `outcome` varchar(225) DEFAULT NULL,
+            `regimen_id` int(11) DEFAULT NULL,
```
Comment (stansL) [inline api/src/main/resources/liquibase.xml:1725]: Looks like there is no explicit creation of the referential integrity things via code, hope it still works

File: omod/src/main/resources/config.xml [hunk 4]
```diff
@@ -176,6 +176,8 @@
 		WaiverType.hbm.xml
 		WaiverTypeRole.hbm.xml
 		EhrHospitalWaiver.hbm.xml
+        PatientRegimen.hbm.xml
+
```
Comment (stansL) [inline omod/src/main/resources/config.xml:183]: I haven't seen the hibernate config file for the Cycle model and it's equivalent config file entry 
```suggestion
Cycle.hbm.xml
```

**Review by stansL — CHANGES_REQUESTED** [review]
Check the included comments, there are missing methods and hibernate config files and their mappings

File: api/src/main/java/org/openmrs/module/hospitalcore/InventoryCommonService.java (hunk 1 above)
Comment (ngarivictor) [reply api/src/main/java/org/openmrs/module/hospitalcore/InventoryCommonService.java]: there was an issue with this commit lemme rectify this

File: api/src/main/resources/liquibase.xml (hunk 3 above)
Comment (ngarivictor) [reply api/src/main/resources/liquibase.xml:1725]: there are alter statements that sort this out

File: api/src/main/java/org/openmrs/module/hospitalcore/impl/InventoryCommonServiceImpl.java [hunk 5]
```diff
@@ -93,18 +93,30 @@ public Regimen updateRegimen(Regimen regimen) throws APIException {
         return dao.updateRegimen(regimen);
     }
 
+    @Override
+    public List<Regimen> getRegimens(boolean voided) {
+        return null;
```
Comment (stansL) [inline api/src/main/java/org/openmrs/module/hospitalcore/impl/InventoryCommonServiceImpl.java:98]:
```suggestion
       return dao.getRegimens(voided);
```

**Review by stansL — APPROVED** [review]

**Pull request merged**